src/Components/Shared/src/WebEventData.cs

```diff
@@ -44,43 +53,34 @@ private WebEventData(int browserRendererId, ulong eventHandlerId, EventFieldInfo
+        private static EventArgs ParseEventArgsJson(ulong eventHandlerId, string eventArgsType, string eventArgsJson)
         {
-            switch (eventArgsType)
+            try
             {
-                case "change":
-                    return DeserializeChangeEventArgs(eventArgsJson);
-                case "clipboard":
-                    return Deserialize<UIClipboardEventArgs>(eventArgsJson);
-                case "drag":
-                    return Deserialize<UIDragEventArgs>(eventArgsJson);
-                case "error":
-                    return Deserialize<UIErrorEventArgs>(eventArgsJson);
-                case "focus":
-                    return Deserialize<UIFocusEventArgs>(eventArgsJson);
-                case "keyboard":
-                    return Deserialize<UIKeyboardEventArgs>(eventArgsJson);
-                case "mouse":
-                    return Deserialize<UIMouseEventArgs>(eventArgsJson);
-                case "pointer":
-                    return Deserialize<UIPointerEventArgs>(eventArgsJson);
-                case "progress":
-                    return Deserialize<UIProgressEventArgs>(eventArgsJson);
-                case "touch":
-                    return Deserialize<UITouchEventArgs>(eventArgsJson);
-                case "unknown":
-                    return EventArgs.Empty;
-                case "wheel":
-                    return Deserialize<UIWheelEventArgs>(eventArgsJson);
-                default:
-                    throw new ArgumentException($"Unsupported value '{eventArgsType}'.", nameof(eventArgsType));
+                return eventArgsType switch
+                {
+                    "change" => DeserializeChangeEventArgs(eventArgsJson),
+                    "clipboard" => Deserialize<UIClipboardEventArgs>(eventArgsJson),
+                    "drag" => Deserialize<UIDragEventArgs>(eventArgsJson),
+                    "error" => Deserialize<UIErrorEventArgs>(eventArgsJson),
+                    "focus" => Deserialize<UIFocusEventArgs>(eventArgsJson),
+                    "keyboard" => Deserialize<UIKeyboardEventArgs>(eventArgsJson),
+                    "mouse" => Deserialize<UIMouseEventArgs>(eventArgsJson),
+                    "pointer" => Deserialize<UIPointerEventArgs>(eventArgsJson),
+                    "progress" => Deserialize<UIProgressEventArgs>(eventArgsJson),
+                    "touch" => Deserialize<UITouchEventArgs>(eventArgsJson),
+                    "unknown" => EventArgs.Empty,
+                    "wheel" => Deserialize<UIWheelEventArgs>(eventArgsJson),
+                    _ => throw new InvalidOperationException($"Unsupported event type '{eventArgsType}'. EventId: '{eventHandlerId}'."),
+                };
+            }
+            catch (Exception e)
+            {
+                throw new InvalidOperationException($"There was an error parsing the event arguments. EventId: '{eventHandlerId}'.", e);
```

Comment:
This is kinda bizzare, why does including the event id here help anyone track down a problem? ids aren't really memorable.

Reply:
That's fair, but it can give you hints, for example, low value event ids signal that the problem happens relatively early in the lifecycle of the app.

Reply:
It's also you are likely be able to search for in the frames under the network tab, if you want to see the payload.